### DISCUSSION TITLE
Log in with different tenant

### DISCUSSION THREAD

**Discussion by simonkurtz-MSFT** [Azure/azure-dev#1417] [Q&A] [answered]
I see that I can pass `tenant-id` when logging in but it also requires `client-id`. What if I just want to personally log into a different tenant, not using a service principal? How can I do that, please? 

Running `azd version 0.5.0-beta.3 (commit 1c23678aade587d5b9685a256a3c1d2d7f4b42e0)`

**Comment by vhvb1989** [comment]
@simonkurtz-MSFT have you tried this:
- before running `azd login` make sure that the browser that you will use to login has a credential logged in to the tenant you want to use. You can do this by logging in to the Azure Portal and then using the `directory` option to select and move the tenant you want.
- Then run `azd login`.  Select the credential from the browser and that should make ADD to return the correct tenant to azd.

Let me know if this works for you

**Reply by simonkurtz-MSFT** [reply]
@vhvb1989, thank you for your prompt reply. A few months ago I obtained a new subscription due to the general tenant move to "Microsoft Non-Production"; however, my old subscription is my home organization, "Microsoft." When logging in to Azure and explicitly selecting the new subscription where I want to use azd, I'm all set there; however, azd login does not honor that and forces the now read-only home organization's subscription. 

I'm looking for a way to say `azd login --tenant-id <tenant>` and then use my credentials, similarly to how the [Az CLI Login](https://learn.microsoft.com/en-us/cli/azure/authenticate-azure-cli#sign-in-with-a-different-tenant) does it.

**Reply by ellismg** [reply]
@simonkurtz-MSFT We are working on a set of changes such that when you log in with `azd`, `azd` will fetch every tenant that you have access to (not just the accounts home tenant) and then pull subscriptions from each tenant that you have access to (similar to what `az` is doing when you log in).  If we did that, I think it would remove the need to explicitly log into a tenant, but I wanted to see if that matches your understanding or if there's something more going on here.

When using the `az` CLI directly with your account, do you always have to pass `--tenant` or does do the right thing out of the box?  Just trying to understand if the work we have in flight will address this or if there's more to be done (and would force our hand on adding support for `--tenant-id` for user accounts)

**Reply by simonkurtz-MSFT** [reply]
Hi @ellismg, I concur that the approach to pull the tenant would be even better. 

I don't recall having issues with `az login` keeping track of my preferred tenant after I logged in with it the first time. Maybe it's sticky, maybe it's aligned with what @vhvb1989 had mentioned previously. 

Thanks so much for all the work on azd!

**Reply by ellismg** [reply]
> I don't recall having issues with az login keeping track of my preferred tenant after I logged in with it the first time. Maybe it's sticky, maybe it's aligned with what @vhvb1989 had mentioned previously.

It stores a "home tenant id" for every subscription and then uses that tenant when it tries to access the subscription, so that's how it appears "sticky".

**Reply by simonkurtz-MSFT** [reply]
Ah, very smart. Good to know that! =)

**Reply by simonkurtz-MSFT** [reply]
Hi @ellismg, just checking in to see if there has been progress on this. Thank you so much!

**Reply by rajeshkamal5050** [reply]
Hi @simonkurtz-MSFT team is actively working on getting a fix out for this soon.

**Reply by Jaseemkh** [reply]
Hi @rajeshkamal5050  any update on this.I am running into exact same problem :(

**Reply by rajeshkamal5050** [reply]
@Jaseemkh this should have bee fixed with our new releases 0.7.

Could you try with the new bits and lets us know?

**Reply by peterlil** [reply]
Hi @rajeshkamal5050 , I'm having the exact same issue (Microsoft Non-Production) and running azd version 0.8.0-beta.2. I don't see any subscriptions from other tenants than my home tenant.

**Reply by peterlil** [reply, marked as answer]
Nvm, guess the --tenant-id was implemented to work with this instead.
